horaedb/metric_engine/src/manifest.rs

```diff
@@ -184,6 +203,264 @@ impl Manifest {
     }
 }
 
+#[repr(packed)]
+#[derive(Debug)]
+struct SnapshotHeader {
+    pub magic: u32,
+    pub version: u8,
+    pub flag: u8,
+    pub length: u64,
+}
+
+impl SnapshotHeader {
+    // format: | magic(u32) | version(u8) |  flags(u8) | length(u64) |
```

Comment:
Could you write this as box comments like here:
https://github.com/apache/horaedb/blob/6cdf2e66a04f11c60c04cacae0111a14d344a501/src/components/codec/src/columnar/bytes.rs#L34-L36